### PR TITLE
feat(GA): add new resource listener

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

add new resource listener

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new resource listener
2. fix a small problem in accelerator
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/ga' TESTARGS='-run TestAccListener_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ga -v -run TestAccListener_basic -timeout 360m -parallel 4
=== RUN   TestAccListener_basic
=== PAUSE TestAccListener_basic
=== CONT  TestAccListener_basic
--- PASS: TestAccListener_basic (72.59s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ga        72.700s
```

### CODE REVIEW

**Comment by Jason-Zhang9309** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/ga' TESTARGS='-run TestAccListener_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ga -v -run TestAccListener_basic -timeout 360m -parallel 4
=== RUN   TestAccListener_basic
=== PAUSE TestAccListener_basic
=== CONT  TestAccListener_basic
--- PASS: TestAccListener_basic (73.64s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ga        73.765s
```

File: huaweicloud/services/acceptance/ga/resource_huaweicloud_ga_listener_test.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+	"strings"
+	"testing"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/ga/resource_huaweicloud_ga_listener_test.go]:
```suggestion
	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"

	"github.com/chnsz/golangsdk"
	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
```

Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/ga/resource_huaweicloud_ga_listener_test.go]: fixed

**Pull request merged**